cpp/src/strings/repeat_strings.cu

```diff
@@ -172,22 +199,259 @@ std::unique_ptr<column> repeat_strings(strings_column_view const& input,
                              mr);
 }
 
+namespace {
+constexpr auto invalid_size = std::numeric_limits<size_type>::lowest();
+constexpr auto max_offset   = static_cast<int64_t>(std::numeric_limits<size_type>::max());
+
+template <typename Func>
+std::tuple<std::unique_ptr<column>, size_type, rmm::device_buffer, size_type>
+generate_offsets_and_null_mask(Func& fn,
+                               size_type strings_count,
+                               bool generate_null_mask,
+                               check_output_overflow overflow_checking,
+                               rmm::cuda_stream_view stream,
+                               rmm::mr::device_memory_resource* mr)
+{
+  auto offsets_column = make_numeric_column(
+    data_type{type_id::INT32}, strings_count + 1, mask_state::UNALLOCATED, stream, mr);
+  auto offsets_view = offsets_column->mutable_view();
+  auto d_offsets    = offsets_view.template data<int32_t>();
+  fn.d_offsets      = d_offsets;
+
+  auto validities = rmm::device_uvector<int8_t>(generate_null_mask ? strings_count : 0, stream);
+  fn.d_validities = generate_null_mask ? validities.begin() : nullptr;
```

Comment:
Why not create the bitmask array here instead of a temporary int8 column? And just set the bitmask values in the kernel. Otherwise, I recommend using the `valid_if` utility to create the bitmask separately -- since a kernel is needed to convert int8 into bitmask at the end anyway.

Reply:
Ah thanks for this. I didn't know that.

Reply:
Actually, there is a [`cudf::bitmask_and`](https://docs.rapids.ai/api/libcudf/stable/group__column__nullmask.html#gad02d632cd9889f9ad12b706b4221f81f) operation that can be used here. Since the output should be an AND of the `input` bitmask and the `repeat_times` bitmask.